api/views.py

```diff
@@ -1,3 +1,236 @@
+class ParticipantJoinTeamView(generics.UpdateAPIView):
+    queryset = models.Team.objects.all()
+    serializer_class = serializers.TeamSerializer
+    lookup_field = 'id'
+
+    def update(self, request, *args, **kwargs):
+        # L'équipe est déjà récupérée grâce à 'get_object()'
+        team = self.get_object()
+
+        # Ajouter le participant (le reste de la logique est identique)
+        user = request.user
+        try:
+            participant = models.Participant.objects.get(user=user)
+        except models.Participant.DoesNotExist:
+            return Response({'detail': 'Participant not found.'}, status=status.HTTP_404_NOT_FOUND)
+
+        if participant in team.participants.all():
+            return Response({'detail': 'You are already in this team.'}, status=status.HTTP_400_BAD_REQUEST)
+
+        # Ajouter le participant
+        team.participants.add(participant)
+
+        # Vérifier si l'équipe est pleine
+        if team.participants.count() >= team.hackaton.teamLimit:
+            team.full = True
+            team.save()
+
+        # Sauvegarder l'équipe après modification
+        team.save()
+
+        return Response(
+            {'message': f"Team {team.name} {team.code} updated successfully. Participant {user.first_name} joined."},
+            status=status.HTTP_200_OK
+        )
+
+class ParticipantLeaveTeamView(generics.UpdateAPIView):
+    queryset = models.Team.objects.all()
+    serializer_class = serializers.TeamSerializer
+    lookup_field = 'id'
+
+    def update(self, request, *args, **kwargs):
+        # L'équipe est déjà récupérée grâce à 'get_object()'
+        team = self.get_object()
+
+        # Ajouter le participant (le reste de la logique est identique)
+        user = request.user
+        try:
+            participant = models.Participant.objects.get(user=user)
+        except models.Participant.DoesNotExist:
+            return Response({'detail': 'Participant not found.'}, status=status.HTTP_404_NOT_FOUND)
+
+        if participant not in team.participants.all():
+            return Response({'detail': 'You are not in this team.'}, status=status.HTTP_400_BAD_REQUEST)
+
+        # Ajouter le participant
+        team.participants.remove(participant)
+
+        # Vérifier si l'équipe est pleine
+        if team.participants.count() < team.hackaton.teamLimit:
+            team.full = False
+            team.save()
+
+        # Sauvegarder l'équipe après modification
+        team.save()
+
+        return Response(
+            {'message': f"Team {team.name} {team.code} updated successfully. Participant {user.first_name} left."},
+            status=status.HTTP_200_OK
+        )
+
+# class Hackaton
+class HackatonListView(generics.ListAPIView):
+    permission_classes = [IsAuthenticated]  # Permet d'assurer que l'utilisateur est authentifié
+    queryset = models.Hackaton.objects.all()
+    serializer_class = serializers.HackatonSerializer
+
+class HackatonCreateView(generics.CreateAPIView):
```

Comment:
you could've used the ListCreateAPIVIEW instead of having two different views

Reply:
ok i'll see how to do it with this generic